src/app-layer-parser.c

```diff
@@ -1275,7 +1278,7 @@ static void HandleStreamFrames(Flow *f, StreamSlice stream_slice, const uint8_t
     } else if (flags & STREAM_EOF) {
         Frame *frame = AppLayerFrameGetById(f, direction, FRAME_STREAM_ID);
         SCLogDebug("EOF closing: frame %p", frame);
-        if (frame) {
+        if (frame && frame->type == FRAME_STREAM_TYPE) {
```

Comment:
what happens to the stream frame if this is false?

Reply:
> what happens to the stream frame if this is false?

Do you mean by `if this is false` like `if (frame && frame->type != FRAME_STREAM_TYPE) `, that is we have a first frame, but it is not a stream frame.

In this case, the stream frame was not created, meaning that the call of `AppLayerFrameNewByAbsoluteOffset` returned NULL because of `FrameConfigTypeIsEnabled` finding that this frame is not used in rules

That happened on the `http2-frames` SV test

> it would imply frame id 1 is not the stream frame, so we'd have to look for it to close it?

Frame 1 is not the stream frame, the stream frame does not exist because it is unneeded...


Does that answer your question ?
If so, how would you improve my commit message ?

Reply:
What confused me is "stream frame is not always the first frame", suggesting, to me, that it can also be coming after another frame. But now I read the above, I understand it's meant to say that it's not always created, hence the ID 1 can also be for a non-stream frame. Is my understanding correct? If so, I think this point should be made more clear in the commit.

Reply:
>  Is my understanding correct? 

Oh yes

> "stream frame is not always the first frame"

Should rather be said this way "the first frame is not always the stream frame" ?

Reply:
Considering the explanations and conversation here, can the commit message explain something along the lines of:
Sometimes, the stream frame doesn't exist, as it is not needed.
In such cases, the first frame (ID 1) will not be a stream frame, but some other enabled frame...

?

Reply:
Thank you Juliana, did you see the new PR ? Tell me if it still needs clarity